### PR TITLE
fix: incorrect show create table output

### LINKED ISSUE #1459: bug: incorrect show create table output
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Frontend

### What happened?

When we try to execute the `show create table` command on an external table, it prints the following:
```
mysql> show create table vt_json;
+---------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table   | Create Table                                                                                                                                                                                                                                                                                                                                                                                                       |
+---------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| vt_json | CREATE TABLE IF NOT EXISTS vt_json (
  a BIGINT NULL,
  b DOUBLE NULL,
  c BOOLEAN NULL,
  d STRING NULL,
  e TIMESTAMP(0) NULL,
  f DOUBLE NULL,
  g TIMESTAMP(0) NULL,

)
ENGINE=file
WITH(
  FORMAT = 'json',
  LOCATION = '/Users/wenyxu/Projects/greptimedb/tests/data/json/various_type.json',
  IMMUTABLE_TABLE_META = '{"files":["Users/wenyxu/Projects/greptimedb/tests/data/json/various_type.json"]}'
) |
+---------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

The correct output should be :

```
mysql> show create external table vt_json;
+---------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table   | Create Table                                                                                                                                                                                                                                                                                                                                                                                                       |
+---------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| vt_json | CREATE TABLE IF NOT EXISTS vt_json (
  a BIGINT NULL,
  b DOUBLE NULL,
  c BOOLEAN NULL,
  d STRING NULL,
  e TIMESTAMP(0) NULL,
  f DOUBLE NULL,
  g TIMESTAMP(0) NULL,

)
ENGINE=file
WITH(
  FORMAT = 'json',
  LOCATION = '/Users/wenyxu/Projects/greptimedb/tests/data/json/various_type.json',
) |
+---------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

### What operating system did you use?

macOS 13.3.1

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

```
create external table vt_csv with (location='/Users/wenyxu/Projects/greptimedb/tests/data/csv/various_type.csv',format='csv');

show create table vt_csv;
```

### PR DESCRIPTION
**Pull request by NiwakaDev**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

When running `show create table` on an external table, this PR fixes `greptimedb` showing `IMMUTABLE_TABLE_META`.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

close #1459

### CODE REVIEW

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Review by WenyXu — COMMENTED** [review]
Maybe we can insert `EXTERNAL` here by checking whether the `engine==IMMUTABLE_FILE_ENGINE`
https://github.com/GreptimeTeam/greptimedb/blob/5cb9adf402da0b9eb4700f604cfe7d4d7c069358/src/sql/src/statements/create.rs#L159C1-L179

File: src/sql/src/statements/create.rs [hunk 1]
```diff
@@ -110,7 +111,8 @@ impl CreateTable {
         if self.options.is_empty() {
             "".to_string()
         } else {
-            let options = format_list_indent!(self.options);
+            let options: Vec<&SqlOption> = self.options.iter().sorted().collect();
```
Comment (NiwakaDev) [inline src/sql/src/statements/create.rs:114]: `test_show_create_external_table_sql` sometimes fails due to `self.options` not being sorted. So, I appended this.

File: src/query/src/sql/show.rs [hunk 2]
```diff
@@ -74,7 +75,11 @@ fn create_sql_options(table_meta: &TableMeta) -> Vec<SqlOption> {
         options.push(sql_option("compaction_time_window", number_value(w)));
     }
 
-    for (k, v) in &table_opts.extra_options {
+    for (k, v) in table_opts
+        .extra_options
+        .iter()
+        .filter(|(k, _)| k != &IMMUTABLE_TABLE_META_KEY)
+    {
         options.push(sql_option(k, string_value(v)));
```
Comment (NiwakaDev) [inline src/query/src/sql/show.rs:83]: The order in which `FORMAT` and `LOCATION` are retrieved from the iterator is not deterministic. So, I sorted `self.options` in the `src/sql/src/statements/create.rs`

File: tests/cases/standalone/show/show_create.result [hunk 3]
```diff
@@ -32,8 +32,8 @@ SHOW CREATE TABLE system_metrics;
 |                | ENGINE=mito                                             |
 |                | WITH(                                                   |
 |                |   regions = 1,                                          |
-|                |   write_buffer_size = '1.0KiB',                         |
-|                |   ttl = '7days'                                         |
+|                |   ttl = '7days',                                        |
+|                |   write_buffer_size = '1.0KiB'                          |
 |                | )                                                       |
 +----------------+---------------------------------------------------------+
```
Comment (NiwakaDev) [inline tests/cases/standalone/show/show_create.result:39]: This is changed by sorting `self.options`.

**Review by WenyXu — APPROVED** [review]
Looks good to me, Thanks @NiwakaDev 🙌

File: src/sql/src/statements/create.rs (hunk 1 above)
Comment (waynexia) [reply src/sql/src/statements/create.rs:114]: This looks great. Stable ordering is also helpful for testing

**Review by waynexia — APPROVED** [review]
LGTM, thanks

**Pull request merged**